lib/core/src/display/world/scene.rs

```diff
@@ -23,9 +23,11 @@ pub struct Scene {
 // === Implementation ===
 
 impl Scene {
-    pub fn new(logger:Logger, globals:&UniformScope) -> Self {
-        let root   = DisplayObjectData::new(logger.sub("root"));
-        let camera = Camera2d::new(logger.sub("camera"),globals);
+    pub fn new(logger:Logger, width:f32, height:f32, globals:&UniformScope) -> Self {
+        let root    = DisplayObjectData::new(logger.sub("root"));
+        let camera  = Camera2d::new(logger.sub("camera"),width,height);
+        let uniform = globals.add_or_panic("zoom", 1.0);
+        camera.add_zoom_update_callback(move |zoom| uniform.set(zoom));
```

Comment:
This uniform is not stored, it is destoryed when going out of this scope. The fact that it is working is a bug. You should stroe unifoirm and assume that when it gous out of scope it will be removed.

Reply:
@wdanilo is it wrong to assume that it will live as long as the closure lives?

Reply:
Oh, good point. Hmm, the only problem is that if we will ever replace camera (render from 2 or more cameras), which we will, then we will to handle this in special case and this uniform will cause problems. It will be better to just store it in the scene.